src/nnet3/decodable-simple-looped.cc

```diff
@@ -86,10 +85,8 @@ void DecodableNnetSimpleLoopedInfo::Init(
   CompileLooped(*nnet, opts.optimize_config, request1, request2, request3,
                 &computation);
   computation.ComputeCudaIndexes();
-  if (GetVerboseLevel() >= 3) {
-    KALDI_VLOG(3) << "Computation is:";
-    computation.Print(std::cerr, *nnet);
-  }
+  KALDI_VLOG(3) << "Computation is:\n"
+                << NnetComputationPrintInserter{computation, *nnet};
```

Comment:
I think the reason I did this to cerr directly rather than to a temporary string is that it is in danger of exhausting memory otherwise, it's very verbose.   I believe these macros use temporary strings and store the string somewhere... I don't think this PR is a good idea, for that reason.

Reply:
Just checked--about 45K characters, which is pretty much nothing. Actually, this is a cherry-picked changeset that has been in production here since January, 21 last year, if I'm to trust my Git. And we valgrind it thoroughly.

The string is deleted when the MessageLogger goes out of scope, as soon as this line ends executing; it's a constructed temporary without a name.

Reply:
You did actually declare the `operator<<` (without a definition), but printing requires both the computation and the nnet, so requires an inserter. It really looked like something left to implement later. So I just caught up where you left it : )